Core/HLE/proAdhoc.cpp

```diff
@@ -1296,11 +1304,26 @@ int getLocalIp(sockaddr_in * SocketAddress){
 	pHost = ::gethostbyname(szHostName);
 	if(pHost) {
 		memcpy(&SocketAddress->sin_addr, pHost->h_addr_list[0], pHost->h_length);
+		if (isLocalServer) {
+			SocketAddress->sin_addr = ((sockaddr_in*)&localIP)->sin_addr;
+		}
 		return 0;
 	}
 	return -1;
 #else
 	memcpy(&SocketAddress->sin_addr, &localip, sizeof(uint32_t));
+	char szHostName[256] = "";
+	gethostname(szHostName, sizeof(szHostName));
+	struct hostent* pHost = 0;
+	pHost = gethostbyname(szHostName);
+	if (pHost) {
+		memcpy(&SocketAddress->sin_addr, pHost->h_addr_list[0], pHost->h_length);
+		if (isLocalServer) {
+			SocketAddress->sin_addr = ((sockaddr_in*)&localIP)->sin_addr;
+		}
+		return 0;
+	}
+	//return -1;
```

Comment:
I prefer to return -1 (failed) if the address didn't passed the validity test tho, but since i'm not trying to be in the way of the person who wrote the previous code so i just commented my code :D just in case that person have plans for his code in the future